<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>What is Hotgenes? • Hotgenes</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="What is Hotgenes?">
<meta property="og:description" content="Hotgenes">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Hotgenes</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.0.25</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/Hotgenes.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>What is Hotgenes?</h1>
            
      
      
      <div class="hidden name"><code>Hotgenes.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="hotgenes-is-an-r-package-built-to-simplify-analysis-of-omics-data-">Hotgenes is an R package built to simplify analysis of omics
data.<a class="anchor" aria-label="anchor" href="#hotgenes-is-an-r-package-built-to-simplify-analysis-of-omics-data-"></a>
</h2>
</div>
<div class="section level2">
<h2 id="convert-deseq2-analysis-into-a-hotgenes-object">Convert DESeq2 analysis into a Hotgenes Object<a class="anchor" aria-label="anchor" href="#convert-deseq2-analysis-into-a-hotgenes-object"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span> <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va">Hotgenes</span><span class="op">)</span></span>
<span><span class="co">## Loading required package: Hotgenes</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span>  <span class="co"># incase you wanted to include aliases for your genes</span></span>
<span>  <span class="co"># requires a "Feature" column that contains gene names in expression matrix</span></span>
<span>  <span class="va">dbCon</span> <span class="op">&lt;-</span> <span class="fu">org.Hs.eg.db</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/org.Hs.eg.db/man/org.Hs.eg_dbconn.html" class="external-link">org.Hs.eg_dbconn</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">## </span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>  <span class="va">sqlQuery</span> <span class="op">&lt;-</span> <span class="st">"SELECT * FROM ENSEMBL, gene_info WHERE ENSEMBL._id == gene_info._id;"</span></span>
<span></span>
<span>  <span class="va">ensembl_Symbol</span> <span class="op">&lt;-</span> <span class="fu">DBI</span><span class="fu">::</span><span class="fu"><a href="https://dbi.r-dbi.org/reference/dbGetQuery.html" class="external-link">dbGetQuery</a></span><span class="op">(</span><span class="va">dbCon</span>, <span class="va">sqlQuery</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Feature"</span> <span class="op">=</span> <span class="st">"symbol"</span>, <span class="st">"ensembl_id"</span><span class="op">)</span><span class="op">)</span></span>
<span>    </span>
<span></span>
<span></span>
<span>  <span class="co"># HotgenesDEseq2 ----------------------------------------------------------</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://github.com/thelovelab/DESeq2" class="external-link">DESeq2</a></span><span class="op">)</span></span>
<span><span class="co">## Loading required package: DESeq2</span></span>
<span><span class="co">## Loading required package: S4Vectors</span></span>
<span><span class="co">## Loading required package: stats4</span></span>
<span><span class="co">## Loading required package: BiocGenerics</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'BiocGenerics'</span></span>
<span><span class="co">## The following objects are masked from 'package:stats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     IQR, mad, sd, var, xtabs</span></span>
<span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     anyDuplicated, aperm, append, as.data.frame, basename, cbind,</span></span>
<span><span class="co">##     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,</span></span>
<span><span class="co">##     get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,</span></span>
<span><span class="co">##     match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,</span></span>
<span><span class="co">##     Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,</span></span>
<span><span class="co">##     table, tapply, union, unique, unsplit, which.max, which.min</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'S4Vectors'</span></span>
<span><span class="co">## The following object is masked from 'package:utils':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     findMatches</span></span>
<span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     expand.grid, I, unname</span></span>
<span><span class="co">## Loading required package: IRanges</span></span>
<span><span class="co">## Loading required package: GenomicRanges</span></span>
<span><span class="co">## Loading required package: GenomeInfoDb</span></span>
<span><span class="co">## Loading required package: SummarizedExperiment</span></span>
<span><span class="co">## Loading required package: MatrixGenerics</span></span>
<span><span class="co">## Loading required package: matrixStats</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'MatrixGenerics'</span></span>
<span><span class="co">## The following objects are masked from 'package:matrixStats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,</span></span>
<span><span class="co">##     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,</span></span>
<span><span class="co">##     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,</span></span>
<span><span class="co">##     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,</span></span>
<span><span class="co">##     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,</span></span>
<span><span class="co">##     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,</span></span>
<span><span class="co">##     colWeightedMeans, colWeightedMedians, colWeightedSds,</span></span>
<span><span class="co">##     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,</span></span>
<span><span class="co">##     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,</span></span>
<span><span class="co">##     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,</span></span>
<span><span class="co">##     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,</span></span>
<span><span class="co">##     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,</span></span>
<span><span class="co">##     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,</span></span>
<span><span class="co">##     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,</span></span>
<span><span class="co">##     rowWeightedSds, rowWeightedVars</span></span>
<span><span class="co">## Loading required package: Biobase</span></span>
<span><span class="co">## Welcome to Bioconductor</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     Vignettes contain introductory material; view with</span></span>
<span><span class="co">##     'browseVignettes()'. To cite Bioconductor, see</span></span>
<span><span class="co">##     'citation("Biobase")', and for packages 'citation("pkgname")'.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'Biobase'</span></span>
<span><span class="co">## The following object is masked from 'package:MatrixGenerics':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     rowMedians</span></span>
<span><span class="co">## The following objects are masked from 'package:matrixStats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     anyMissing, rowMedians</span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>  </span>
<span>  <span class="co"># loading example data</span></span>
<span>  <span class="va">dds_con_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>,</span>
<span>    <span class="st">"dds_con.Rdata"</span>,</span>
<span>    package <span class="op">=</span> <span class="st">"Hotgenes"</span>,</span>
<span>    mustWork <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>  <span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="va">dds_con_dir</span><span class="op">)</span></span>
<span></span>
<span>  <span class="co"># Example Expression data and coldata</span></span>
<span>  <span class="va">cts</span> <span class="op">&lt;-</span> <span class="fu">DESeq2</span><span class="fu">::</span><span class="fu">counts</span><span class="op">(</span><span class="va">dds_con</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="va">Design</span> <span class="op">&lt;-</span> <span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">dds_con</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select_all.html" class="external-link">select_if</a></span><span class="op">(</span><span class="va">is.factor</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Time <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Factor-class.html" class="external-link">levels</a></span><span class="op">(</span><span class="va">.data</span><span class="op">$</span><span class="va">Hrs</span><span class="op">)</span><span class="op">)</span><span class="op">[</span><span class="va">.data</span><span class="op">$</span><span class="va">Hrs</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span>  <span class="co"># set up DESeq2</span></span>
<span>  <span class="va">model_DESeq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/eval.html" class="external-link">eval</a></span><span class="op">(</span><span class="op">~</span> <span class="va">sh</span> <span class="op">*</span> <span class="va">Hrs</span><span class="op">)</span></span>
<span></span>
<span>  <span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu">DESeq2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html" class="external-link">DESeqDataSetFromMatrix</a></span><span class="op">(</span></span>
<span>    countData <span class="op">=</span> <span class="va">cts</span>,</span>
<span>    colData <span class="op">=</span> <span class="va">Design</span>,</span>
<span>    design <span class="op">=</span> <span class="va">model_DESeq</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span>  <span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu">DESeq2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html" class="external-link">DESeq</a></span><span class="op">(</span><span class="va">dds</span><span class="op">)</span></span>
<span><span class="co">## estimating size factors</span></span>
<span><span class="co">## estimating dispersions</span></span>
<span><span class="co">## gene-wise dispersion estimates</span></span>
<span><span class="co">## mean-dispersion relationship</span></span>
<span><span class="co">## final dispersion estimates</span></span>
<span><span class="co">## fitting model and testing</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span>  <span class="co"># Convert to Hotgenes Object</span></span>
<span></span>
<span>  <span class="va">dds_Hotgenes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/HotgenesDEseq2.html">HotgenesDEseq2</a></span><span class="op">(</span></span>
<span>    DEseq2_object <span class="op">=</span> <span class="va">dds</span>,</span>
<span>    lfcShrink_type <span class="op">=</span> <span class="st">"apeglm"</span>,</span>
<span>    contrasts <span class="op">=</span> <span class="st">"sh_EWS_vs_Ctrl"</span>, </span>
<span>    ExpressionData <span class="op">=</span> <span class="st">"vsd"</span>, </span>
<span>    Mapper <span class="op">=</span> <span class="va">ensembl_Symbol</span></span>
<span>  <span class="op">)</span></span>
<span><span class="co">## using 'apeglm' for LFC shrinkage. If used in published research, please cite:</span></span>
<span><span class="co">##     Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for</span></span>
<span><span class="co">##     sequence count data: removing the noise and preserving large differences.</span></span>
<span><span class="co">##     Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="convert-limma-de-analysis-into-hotgenes-object">Convert limma DE analysis into Hotgenes Object<a class="anchor" aria-label="anchor" href="#convert-limma-de-analysis-into-hotgenes-object"></a>
</h2>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span> <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va">Hotgenes</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># incase you wanted to include aliases for your genes</span></span>
<span>  <span class="co"># requires a "Feature" column that contains gene names in expression matrix</span></span>
<span>  <span class="va">dbCon</span> <span class="op">&lt;-</span> <span class="fu">org.Hs.eg.db</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/org.Hs.eg.db/man/org.Hs.eg_dbconn.html" class="external-link">org.Hs.eg_dbconn</a></span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="va">sqlQuery</span> <span class="op">&lt;-</span></span>
<span>    <span class="st">"SELECT * FROM ENSEMBL, gene_info WHERE ENSEMBL._id == gene_info._id;"</span></span>
<span>  </span>
<span>  <span class="va">ensembl_Symbol</span> <span class="op">&lt;-</span> <span class="fu">DBI</span><span class="fu">::</span><span class="fu"><a href="https://dbi.r-dbi.org/reference/dbGetQuery.html" class="external-link">dbGetQuery</a></span><span class="op">(</span><span class="va">dbCon</span>, <span class="va">sqlQuery</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Feature"</span> <span class="op">=</span> <span class="st">"symbol"</span>, <span class="st">"ensembl_id"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span>  </span>
<span>  </span>
<span>  </span>
<span>  <span class="co"># Hotgeneslimma -----------------------------------------------------------</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://github.com/thelovelab/DESeq2" class="external-link">DESeq2</a></span><span class="op">)</span></span>
<span></span>
<span>  <span class="va">dds_con_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>,</span>
<span>                             <span class="st">"dds_con.Rdata"</span>,</span>
<span>                             package <span class="op">=</span> <span class="st">"Hotgenes"</span>,</span>
<span>                             mustWork <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="va">dds_con_dir</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># Example Expression data and coldata</span></span>
<span>  <span class="va">cts</span> <span class="op">&lt;-</span> <span class="fu">counts</span><span class="op">(</span><span class="va">dds_con</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="va">Design</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">dds_con</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select_all.html" class="external-link">select_if</a></span><span class="op">(</span><span class="va">is.factor</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Time <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Factor-class.html" class="external-link">levels</a></span><span class="op">(</span><span class="va">.data</span><span class="op">$</span><span class="va">Hrs</span><span class="op">)</span><span class="op">)</span><span class="op">[</span><span class="va">.data</span><span class="op">$</span><span class="va">Hrs</span><span class="op">]</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># Create DGEList object</span></span>
<span>  <span class="co"># and calculate normalization factors</span></span>
<span>  <span class="va">d0</span> <span class="op">&lt;-</span> <span class="fu">edgeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/DGEList.html" class="external-link">DGEList</a></span><span class="op">(</span><span class="va">cts</span><span class="op">)</span></span>
<span>  <span class="va">d0</span> <span class="op">&lt;-</span> <span class="fu">edgeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/calcNormFactors.html" class="external-link">calcNormFactors</a></span><span class="op">(</span><span class="va">d0</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># Filter low-expressed genes</span></span>
<span>  <span class="co"># disabled in this example</span></span>
<span>  <span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">cutoff</span> <span class="op">&lt;-</span> <span class="fl">1</span></span>
<span>    <span class="va">drop</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="fu">cpm</span><span class="op">(</span><span class="va">d0</span><span class="op">)</span>, <span class="fl">1</span>, <span class="va">max</span><span class="op">)</span> <span class="op">&lt;</span> <span class="va">cutoff</span><span class="op">)</span></span>
<span>    <span class="va">d</span> <span class="op">&lt;-</span> <span class="va">d0</span><span class="op">[</span><span class="op">-</span><span class="va">drop</span>,<span class="op">]</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">d</span><span class="op">)</span> <span class="co"># number of genes lef</span></span>
<span>  <span class="op">}</span></span>
<span>  </span>
<span>  <span class="va">d</span> <span class="op">&lt;-</span> <span class="va">d0</span></span>
<span>  </span>
<span>  <span class="co"># make a model.matrix</span></span>
<span>  <span class="va">model_Matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix</a></span><span class="op">(</span> <span class="op">~</span> <span class="va">sh</span> <span class="op">*</span> <span class="va">Hrs</span>,</span>
<span>                                data <span class="op">=</span> <span class="va">Design</span><span class="op">)</span></span>
<span>  <span class="co"># voom</span></span>
<span>  <span class="va">vm_exp</span> <span class="op">&lt;-</span> <span class="fu">limma</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/voom.html" class="external-link">voom</a></span><span class="op">(</span><span class="va">d</span>, <span class="va">model_Matrix</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># make fit</span></span>
<span>  <span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu">limma</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/lmFit.html" class="external-link">lmFit</a></span><span class="op">(</span><span class="va">vm_exp</span>, <span class="va">model_Matrix</span><span class="op">)</span></span>
<span>  <span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu">limma</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">eBayes</a></span><span class="op">(</span><span class="va">fit</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># Get alternative exps</span></span>
<span>  <span class="va">alt_Exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.matrix.html" class="external-link">data.matrix</a></span><span class="op">(</span><span class="va">d0</span><span class="op">)</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># Convert to Hotgenes Object</span></span>
<span>  <span class="va">fit_Hotgenes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Hotgeneslimma.html">Hotgeneslimma</a></span><span class="op">(</span></span>
<span>    limmafit <span class="op">=</span> <span class="va">fit</span>,</span>
<span>    coldata <span class="op">=</span> <span class="va">Design</span>,</span>
<span>    Expression <span class="op">=</span> <span class="va">vm_exp</span>,</span>
<span>    Expression_name <span class="op">=</span> <span class="st">"logCPM"</span>,</span>
<span>    Exps_list <span class="op">=</span> <span class="va">alt_Exp</span>,</span>
<span>    Mapper <span class="op">=</span> <span class="va">ensembl_Symbol</span></span>
<span>  <span class="op">)</span></span>
<span>  </span></code></pre></div>
</div>
<div class="section level2">
<h2 id="for-other-platforms-you-can-generate-a-hotgenes-object-using-hotgenesuniversal-">For other platforms you can generate a hotgenes object using
HotgenesUniversal().<a class="anchor" aria-label="anchor" href="#for-other-platforms-you-can-generate-a-hotgenes-object-using-hotgenesuniversal-"></a>
</h2>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">Hotgenes</span><span class="op">)</span></span>
<span>  </span>
<span>  </span>
<span>  <span class="co"># load example data -------------------------------------------------------</span></span>
<span>  </span>
<span>  <span class="va">dds_Hotgenes_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>,</span>
<span>                                  <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"dds_Hotgenes"</span>, <span class="st">".RDS"</span><span class="op">)</span>,</span>
<span>                                  package <span class="op">=</span> <span class="st">"Hotgenes"</span>,</span>
<span>                                  mustWork <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>  <span class="op">)</span></span>
<span>  </span>
<span>  <span class="va">htgs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="va">dds_Hotgenes_dir</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># preparing data -----------------------------------------------------------</span></span>
<span>  </span>
<span>  <span class="co"># Getting example named list of DE statistics</span></span>
<span>  <span class="va">NewDE</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Hotgenes-class.html">Output_DE_</a></span><span class="op">(</span><span class="va">htgs</span>, as_list <span class="op">=</span> <span class="cn">TRUE</span>, padj_cut <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># Getting example named list of normalized data</span></span>
<span>  <span class="va">NormlData</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Hotgenes-class.html">Normalized_Data_</a></span><span class="op">(</span><span class="va">htgs</span><span class="op">)</span></span>
<span>   </span>
<span>  <span class="co"># Getting example coldata</span></span>
<span>  <span class="va">ExpColdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Hotgenes-class.html">coldata_</a></span><span class="op">(</span><span class="va">htgs</span><span class="op">)</span></span>
<span></span>
<span>  <span class="co"># Getting example original data object used for DE analysis</span></span>
<span>  <span class="co"># This example was generated from DESeq2</span></span>
<span>  <span class="va">OrigDEObj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Hotgenes-class.html">O_</a></span><span class="op">(</span><span class="va">htgs</span><span class="op">)</span></span>
<span>  <span class="va">OrigDEObj</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">## [1] "DESeqDataSet"</span></span>
<span><span class="co">## attr(,"package")</span></span>
<span><span class="co">## [1] "DESeq2"</span></span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>  </span>
<span>  <span class="co"># Getting example design matrix</span></span>
<span>  <span class="va">DE_design</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Hotgenes-class.html">designMatrix_</a></span><span class="op">(</span><span class="va">htgs</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># Getting example mapper</span></span>
<span>  <span class="va">MapperDF</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Hotgenes-class.html">Mapper_</a></span><span class="op">(</span><span class="va">htgs</span><span class="op">)</span></span>
<span>  </span>
<span>  </span>
<span>  <span class="co"># Converting example objects to hotgenes</span></span>
<span>  <span class="va">Hotgenes_Object</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/HotgenesUniversal.html">HotgenesUniversal</a></span><span class="op">(</span></span>
<span>    Output_DE <span class="op">=</span> <span class="va">NewDE</span>,</span>
<span>    Normalized_Expression <span class="op">=</span> <span class="va">NormlData</span>,</span>
<span>    coldata <span class="op">=</span> <span class="va">ExpColdata</span>,</span>
<span>    Original_Object <span class="op">=</span> <span class="va">OrigDEObj</span>,</span>
<span>    designMatrix <span class="op">=</span> <span class="va">DE_design</span>,</span>
<span>    Mapper <span class="op">=</span> <span class="va">MapperDF</span></span>
<span>  <span class="op">)</span></span>
<span>  </span>
<span>  </span></code></pre></div>
</div>
<div class="section level2">
<h2 id="what-can-you-do-with-a-hotgenes-object">What can you do with a Hotgenes object?<a class="anchor" aria-label="anchor" href="#what-can-you-do-with-a-hotgenes-object"></a>
</h2>
<p>calling a Hotgenes object returns a summary table</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">Hotgenes</span><span class="op">)</span></span>
<span></span>
<span><span class="va">fit_Hotgenes</span></span>
<span><span class="co">## class: Hotgenes </span></span>
<span><span class="co">## Original class/package:  EList/limma</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Differential expression (default thresholds): </span></span>
<span><span class="co">## |contrast       | total|</span></span>
<span><span class="co">## |:--------------|-----:|</span></span>
<span><span class="co">## |Hrs_2_vs_0     |    45|</span></span>
<span><span class="co">## |Hrs_6_vs_0     |    71|</span></span>
<span><span class="co">## |sh_EWS_vs_Ctrl |    51|</span></span>
<span><span class="co">## |shEWS.Hrs2     |     8|</span></span>
<span><span class="co">## |shEWS.Hrs6     |     1|</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Available feature mapping:  Feature, ensembl_id </span></span>
<span><span class="co">## ExpressionSlots:  logCPM, counts </span></span>
<span><span class="co">## Total auxiliary assays:  2 </span></span>
<span><span class="co">## Total samples:  12</span></span></code></pre></div>
<p>Sample metadata</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/Hotgenes-class.html">coldata_</a></span><span class="op">(</span><span class="va">fit_Hotgenes</span><span class="op">)</span></span>
<span><span class="co">##                sh Bio_Rep Hrs Time</span></span>
<span><span class="co">## shCON_0hrs_1 Ctrl       1   0    0</span></span>
<span><span class="co">## shCON_0hrs_2 Ctrl       2   0    0</span></span>
<span><span class="co">## shCON_2hrs_1 Ctrl       1   2    2</span></span>
<span><span class="co">## shCON_2hrs_2 Ctrl       2   2    2</span></span>
<span><span class="co">## shCON_6hrs_1 Ctrl       1   6    6</span></span>
<span><span class="co">## shCON_6hrs_2 Ctrl       2   6    6</span></span>
<span><span class="co">## shEWS_0hrs_1  EWS       1   0    0</span></span>
<span><span class="co">## shEWS_0hrs_2  EWS       2   0    0</span></span>
<span><span class="co">## shEWS_2hrs_1  EWS       1   2    2</span></span>
<span><span class="co">## shEWS_2hrs_2  EWS       2   2    2</span></span>
<span><span class="co">## shEWS_6hrs_1  EWS       1   6    6</span></span>
<span><span class="co">## shEWS_6hrs_2  EWS       2   6    6</span></span></code></pre></div>
<p>Normalized expression data</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/Hotgenes-class.html">ExpressionData_</a></span><span class="op">(</span><span class="va">fit_Hotgenes</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="co">##        shCON_0hrs_1 shCON_0hrs_2 shCON_2hrs_1</span></span>
<span><span class="co">## AGER       5.699205     6.356261     6.030818</span></span>
<span><span class="co">## ALOX12     3.377277     3.372749     3.472823</span></span>
<span><span class="co">## ALOX15     5.699205     5.795961     5.512351</span></span></code></pre></div>
<p>Available aliases</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="../reference/Hotgenes-class.html">Mapper_</a></span><span class="op">(</span><span class="va">fit_Hotgenes</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">## <span style="color: #949494;"># A tibble: 6 × 2</span></span></span>
<span><span class="co">##   Feature ensembl_id     </span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> AGER    ENSG00000204305</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> AGER    ENSG00000230514</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> AGER    ENSG00000229058</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">4</span> AGER    ENSG00000231268</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">5</span> AGER    ENSG00000237405</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">6</span> AGER    ENSG00000234729</span></span></code></pre></div>
<p>Available auxiliary_assays</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="../reference/Hotgenes-class.html">auxiliary_assays_</a></span><span class="op">(</span><span class="va">fit_Hotgenes</span><span class="op">)</span></span>
<span><span class="co">##                  assay1      assay2</span></span>
<span><span class="co">## shCON_0hrs_1 -1.4805676 -0.77956651</span></span>
<span><span class="co">## shCON_0hrs_2  1.5771695  0.01195176</span></span>
<span><span class="co">## shCON_2hrs_1 -0.9567445 -0.15241624</span></span>
<span><span class="co">## shCON_2hrs_2 -0.9200052 -0.70346425</span></span>
<span><span class="co">## shCON_6hrs_1 -1.9976421  1.18887916</span></span>
<span><span class="co">## shCON_6hrs_2 -0.2722960  0.34051227</span></span>
<span><span class="co">## shEWS_0hrs_1 -0.3153487  0.50696817</span></span>
<span><span class="co">## shEWS_0hrs_2 -0.6282552 -0.29330515</span></span>
<span><span class="co">## shEWS_2hrs_1 -0.1064639  0.22364142</span></span>
<span><span class="co">## shEWS_2hrs_2  0.4280148  2.00720146</span></span>
<span><span class="co">## shEWS_6hrs_1 -0.7777196  1.01197912</span></span>
<span><span class="co">## shEWS_6hrs_2 -1.2938823 -0.30245925</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="easy-access-to-features-of-interest">Easy access to features of interest<a class="anchor" aria-label="anchor" href="#easy-access-to-features-of-interest"></a>
</h2>
<div class="section level3">
<h3 id="feature-associated-statistics-">Feature-associated statistics.<a class="anchor" aria-label="anchor" href="#feature-associated-statistics-"></a>
</h3>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="../reference/utilities.html">DE</a></span><span class="op">(</span><span class="va">fit_Hotgenes</span>, Topn <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span>
<span><span class="co">## $Hrs_2_vs_0</span></span>
<span><span class="co">## <span style="color: #949494;"># A tibble: 3 × 11</span></span></span>
<span><span class="co">##   Feature contrast_dir  baseMean log2FoldChange    FC  stat   pvalue     padj</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> IL6     Hrs_2_vs_0_up     10.6           2.52  5.72  31.1 4.06<span style="color: #949494;">e</span><span style="color: #BB0000;">-24</span> 1.04<span style="color: #949494;">e</span><span style="color: #BB0000;">-21</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> CXCL8   Hrs_2_vs_0_up     10.9           3.51 11.4   26.0 6.99<span style="color: #949494;">e</span><span style="color: #BB0000;">-22</span> 8.92<span style="color: #949494;">e</span><span style="color: #BB0000;">-20</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> TNFAIP3 Hrs_2_vs_0_up     10.5           2.14  4.42  24.9 2.21<span style="color: #949494;">e</span><span style="color: #BB0000;">-21</span> 1.88<span style="color: #949494;">e</span><span style="color: #BB0000;">-19</span></span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 3 more variables: t &lt;dbl&gt;, B &lt;dbl&gt;, ensembl_id &lt;chr&gt;</span></span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $Hrs_6_vs_0</span></span>
<span><span class="co">## <span style="color: #949494;"># A tibble: 3 × 11</span></span></span>
<span><span class="co">##   Feature contrast_dir  baseMean log2FoldChange    FC  stat   pvalue     padj</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> CXCL8   Hrs_6_vs_0_up     10.9           3.56 11.8   26.4 4.21<span style="color: #949494;">e</span><span style="color: #BB0000;">-22</span> 1.07<span style="color: #949494;">e</span><span style="color: #BB0000;">-19</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> CXCL1   Hrs_6_vs_0_up     11.3           2.35  5.09  24.5 3.71<span style="color: #949494;">e</span><span style="color: #BB0000;">-21</span> 4.73<span style="color: #949494;">e</span><span style="color: #BB0000;">-19</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> CCL2    Hrs_6_vs_0_up     12.5           1.25  2.37  23.0 2.14<span style="color: #949494;">e</span><span style="color: #BB0000;">-20</span> 1.56<span style="color: #949494;">e</span><span style="color: #BB0000;">-18</span></span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 3 more variables: t &lt;dbl&gt;, B &lt;dbl&gt;, ensembl_id &lt;chr&gt;</span></span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $sh_EWS_vs_Ctrl</span></span>
<span><span class="co">## <span style="color: #949494;"># A tibble: 3 × 11</span></span></span>
<span><span class="co">##   Feature contrast_dir     baseMean log2FoldChange    FC  stat   pvalue     padj</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>               <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> C1R     sh_EWS_vs_Ctrl_…     13.5         -<span style="color: #BB0000;">0.856</span> 0.553 -<span style="color: #BB0000;">16.7</span> 1.33<span style="color: #949494;">e</span><span style="color: #BB0000;">-16</span> 3.39<span style="color: #949494;">e</span><span style="color: #BB0000;">-14</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> C1R     sh_EWS_vs_Ctrl_…     13.5         -<span style="color: #BB0000;">0.856</span> 0.553 -<span style="color: #BB0000;">16.7</span> 1.33<span style="color: #949494;">e</span><span style="color: #BB0000;">-16</span> 3.39<span style="color: #949494;">e</span><span style="color: #BB0000;">-14</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> CCL2    sh_EWS_vs_Ctrl_…     12.5         -<span style="color: #BB0000;">1.00</span>  0.500 -<span style="color: #BB0000;">15.8</span> 5.92<span style="color: #949494;">e</span><span style="color: #BB0000;">-16</span> 7.55<span style="color: #949494;">e</span><span style="color: #BB0000;">-14</span></span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 3 more variables: t &lt;dbl&gt;, B &lt;dbl&gt;, ensembl_id &lt;chr&gt;</span></span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $shEWS.Hrs2</span></span>
<span><span class="co">## <span style="color: #949494;"># A tibble: 3 × 11</span></span></span>
<span><span class="co">##   Feature contrast_dir  baseMean log2FoldChange    FC  stat  pvalue   padj     t</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> TNFAIP3 shEWS.Hrs2_d…    10.5          -<span style="color: #BB0000;">0.565</span> 0.676 -<span style="color: #BB0000;">4.69</span> 5.84<span style="color: #949494;">e</span><span style="color: #BB0000;">-5</span> 0.014<span style="text-decoration: underline;">9</span> -<span style="color: #BB0000;">4.69</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> CXCL3   shEWS.Hrs2_d…     7.90         -<span style="color: #BB0000;">1.05</span>  0.482 -<span style="color: #BB0000;">3.94</span> 4.57<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span> 0.023<span style="text-decoration: underline;">3</span> -<span style="color: #BB0000;">3.94</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> FOS     shEWS.Hrs2_up     8.98          0.645 1.56   4.14 2.67<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span> 0.023<span style="text-decoration: underline;">3</span>  4.14</span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 2 more variables: B &lt;dbl&gt;, ensembl_id &lt;chr&gt;</span></span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $shEWS.Hrs6</span></span>
<span><span class="co">## <span style="color: #949494;"># A tibble: 1 × 11</span></span></span>
<span><span class="co">##   Feature contrast_dir  baseMean log2FoldChange    FC  stat  pvalue   padj     t</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> FOS     shEWS.Hrs6_up     8.98          0.826  1.77  4.53 8.92<span style="color: #949494;">e</span><span style="color: #BB0000;">-5</span> 0.022<span style="text-decoration: underline;">7</span>  4.53</span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 2 more variables: B &lt;dbl&gt;, ensembl_id &lt;chr&gt;</span></span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="summary-plot-of-contrasts">Summary plot of contrasts<a class="anchor" aria-label="anchor" href="#summary-plot-of-contrasts"></a>
</h3>
<p>all comparisons</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="../reference/utilities.html">DEPlot</a></span><span class="op">(</span><span class="va">fit_Hotgenes</span>, .log2FoldChange <span class="op">=</span> <span class="fl">0</span>, padj_cut <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span></span></code></pre></div>
<p><img src="Hotgenes_files/figure-html/DEP_1-1.png" width="576"></p>
<p>Check for a feature of interest</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Check a feature across comparisons</span></span>
<span><span class="fu"><a href="../reference/utilities.html">DEPlot</a></span><span class="op">(</span><span class="va">fit_Hotgenes</span>, hotList <span class="op">=</span> <span class="st">"CSF1"</span>, .log2FoldChange <span class="op">=</span> <span class="fl">0</span>, padj_cut <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span></span></code></pre></div>
<p><img src="Hotgenes_files/figure-html/DEP_2-1.png" width="576"></p>
</div>
<div class="section level3">
<h3 id="volcano-plots">Volcano plots<a class="anchor" aria-label="anchor" href="#volcano-plots"></a>
</h3>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/VPlot.html">VPlot</a></span><span class="op">(</span><span class="va">fit_Hotgenes</span>, </span>
<span>      .log2FoldChange <span class="op">=</span> <span class="fl">1</span>, padj_cut <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>      contrasts <span class="op">=</span> <span class="st">"sh_EWS_vs_Ctrl"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Hotgenes_files/figure-html/VPlot_1-1.png" width="576"></p>
</div>
<div class="section level3">
<h3 id="identify-overlapping-features-across-comparisons-with-venn_report">Identify overlapping features across comparisons with
Venn_Report()<a class="anchor" aria-label="anchor" href="#identify-overlapping-features-across-comparisons-with-venn_report"></a>
</h3>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Venn Diagram plot</span></span>
<span><span class="va">fit_Hotgenes</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>   <span class="fu"><a href="../reference/utilities.html">DE</a></span><span class="op">(</span></span>
<span>      Report <span class="op">=</span> <span class="st">"Features"</span>,</span>
<span>      contrasts <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"sh_EWS_vs_Ctrl"</span>, <span class="st">"Hrs_2_vs_0"</span>, <span class="st">"Hrs_6_vs_0"</span><span class="op">)</span>,</span>
<span>      .log2FoldChange <span class="op">=</span> <span class="fl">0</span>, padj_cut <span class="op">=</span> <span class="fl">0.1</span></span>
<span>    <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/Venn_Report.html">Venn_Report</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">## Coordinate system already present. Adding new coordinate system, which will</span></span>
<span><span class="co">## replace the existing one.</span></span>
<span><span class="co">## $vennD</span></span></code></pre></div>
<p><img src="Hotgenes_files/figure-html/Venn_Report_1-1.png" width="768"></p>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## $Intsect</span></span>
<span><span class="co">## $Intsect$Hrs_2_vs_0</span></span>
<span><span class="co">## [1] "NFE2L2" "KEAP1"  "PDGFA"  "HDAC4"  "OXER1"  "GAPDH"  "MEF2C" </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $Intsect$Hrs_6_vs_0</span></span>
<span><span class="co">##  [1] "CXCL5"  "STAT2"  "NR3C1"  "MAP3K1" "HSPB2"  "MAPK8"  "DAXX"   "MKNK1" </span></span>
<span><span class="co">##  [9] "MAP2K6" "IL1B"   "BCL6"   "TLR3"   "GRB2"   "IL6R"   "IL15"   "CREB1" </span></span>
<span><span class="co">## [17] "IL1RN"  "RELA"   "IFIT3"  "MAP3K5" "TGFB3"  "TGFB2"  "IL1A"   "CCL20" </span></span>
<span><span class="co">## [25] "PGK1"   "MAPK3" </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $Intsect$sh_EWS_vs_Ctrl</span></span>
<span><span class="co">##  [1] "HIF1A"  "C3"     "RAC1"   "GNB1"   "TUBB"   "BCL2L1" "CSF1"   "PTGER3"</span></span>
<span><span class="co">##  [9] "ROCK2"  "MX2"    "HMGN1"  "CLTC"   "GNAQ"   "LY96"   "CD40"   "CFD"   </span></span>
<span><span class="co">## [17] "HRAS"   "RHOA"   "HPRT1"  "TCF4"   "MX1"    "OAS2"   "LTB4R2"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $Intsect$`Hrs_2_vs_0:Hrs_6_vs_0`</span></span>
<span><span class="co">##  [1] "CXCL8"   "TNFAIP3" "CXCL1"   "IL11"    "PTGS2"   "DDIT3"   "IFIT2"  </span></span>
<span><span class="co">##  [8] "TGFBR1"  "MAFF"    "CXCR4"   "MAFK"    "PTGFR"   "FOS"     "MYC"    </span></span>
<span><span class="co">## [15] "RIPK2"   "IL2"     "MAFG"    "CSF2"    "TWIST2"  "IFIT1"   "FLT1"   </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $Intsect$`sh_EWS_vs_Ctrl:Hrs_2_vs_0`</span></span>
<span><span class="co">## [1] "HMGB2"  "MAP3K9" "CEBPB"  "IRF1"  </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $Intsect$`sh_EWS_vs_Ctrl:Hrs_6_vs_0`</span></span>
<span><span class="co">##  [1] "C1R"   "C1S"   "MMP3"  "CXCL6" "STAT1" "PTGS1" "HMGB1" "MASP1" "TRAF2"</span></span>
<span><span class="co">## [10] "IFI44" "CCL7" </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $Intsect$`sh_EWS_vs_Ctrl:Hrs_2_vs_0:Hrs_6_vs_0`</span></span>
<span><span class="co">##  [1] "CCL2"     "IL1R1"    "MEF2D"    "JUN"      "BIRC2"    "CXCL3"   </span></span>
<span><span class="co">##  [7] "MEF2A"    "CXCL2"    "IL6"      "NFKB1"    "SMAD7"    "MAPKAPK2"</span></span>
<span><span class="co">## [13] "RELB"    </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $Names</span></span>
<span><span class="co">## [1] "Hrs_2_vs_0:Hrs_6_vs_0"               </span></span>
<span><span class="co">## [2] "sh_EWS_vs_Ctrl:Hrs_2_vs_0"           </span></span>
<span><span class="co">## [3] "sh_EWS_vs_Ctrl:Hrs_6_vs_0"           </span></span>
<span><span class="co">## [4] "sh_EWS_vs_Ctrl:Hrs_2_vs_0:Hrs_6_vs_0"</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="identify-key-features-using-pca-and-clustering">Identify key features using PCA and clustering<a class="anchor" aria-label="anchor" href="#identify-key-features-using-pca-and-clustering"></a>
</h3>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># run PCA</span></span>
<span><span class="co"># set contrast of choice and metadata variables</span></span>
<span><span class="va">FactoOutput</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/FactoWrapper.html">FactoWrapper</a></span><span class="op">(</span><span class="va">fit_Hotgenes</span>,</span>
<span>  contrasts <span class="op">=</span> <span class="st">"Hrs_6_vs_0"</span>,</span>
<span>  coldata_ids <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Hrs"</span>, <span class="st">"sh"</span><span class="op">)</span>,</span>
<span>  biplot <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># plot</span></span>
<span><span class="va">FactoOutput</span><span class="op">$</span><span class="va">res_PPI_pa_1</span></span></code></pre></div>
<p><img src="Hotgenes_files/figure-html/PCA_2-1.png" width="576"></p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># getting HCPC details</span></span>
<span><span class="va">FactoOutput</span><span class="op">$</span><span class="va">TopTibble</span> <span class="co"># Feature</span></span>
<span><span class="co">## <span style="color: #949494;"># A tibble: 88 × 9</span></span></span>
<span><span class="co">##    Cluster Interpretation       Feature v.test `Mean in category` `Overall mean`</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> 1       Cluster has elevated IFIT2     2.91              10.3            9.89</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> 1       Cluster has elevated DDIT3     2.75              10.4           10.1 </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> 1       Cluster has elevated MAP3K1    2.72              10.0            9.75</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> 1       Cluster has elevated PTGFR     2.65              10.4           10.1 </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> 1       Cluster has elevated PGK1      2.63              15.3           15.2 </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> 1       Cluster has elevated TWIST2    2.51              13.3           13.1 </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> 1       Cluster has elevated IFIT3     2.51               9.09           8.85</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> 1       Cluster has elevated MKNK1     2.34              10.9           10.7 </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> 1       Cluster has elevated IFIT1     2.22              10.8           10.4 </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> 1       Cluster has elevated STAT2     2.20              12.6           12.5 </span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 78 more rows</span></span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 3 more variables: `sd in category` &lt;dbl&gt;, `Overall sd` &lt;dbl&gt;, p.value &lt;dbl&gt;</span></span></span></code></pre></div>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">FactoOutput</span><span class="op">$</span><span class="va">TopGroups</span> <span class="co"># TopGroups</span></span>
<span><span class="co">## <span style="color: #949494;"># A tibble: 1 × 8</span></span></span>
<span><span class="co">##   Cluster Interpretation      Category `Cla/Mod` `Mod/Cla` Global p.value v.test</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>               <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> 1       Cluster has elevat… Hrs=Hrs…       100       100   33.3 0.002<span style="text-decoration: underline;">02</span>   3.09</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="streamlined-expression-plots">Streamlined expression plots<a class="anchor" aria-label="anchor" href="#streamlined-expression-plots"></a>
</h3>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Having metadata embedded with expression data means easier plotting</span></span>
<span><span class="va">yvar</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CSF2"</span>, <span class="st">"IL6"</span><span class="op">)</span></span>
<span><span class="va">xvar</span> <span class="op">&lt;-</span> <span class="st">"Hrs"</span></span>
<span></span>
<span><span class="fu"><a href="../reference/ExpsPlot.html">ExpsPlot</a></span><span class="op">(</span><span class="va">fit_Hotgenes</span>,</span>
<span>  xVar <span class="op">=</span> <span class="va">xvar</span>,</span>
<span>  yVar <span class="op">=</span> <span class="va">yvar</span>,</span>
<span>  fill <span class="op">=</span> <span class="st">"Hrs"</span>,</span>
<span>  boxplot <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="Hotgenes_files/figure-html/exps_plot_1-1.png" width="576"></p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># Subset data on the fly</span></span>
<span><span class="fu"><a href="../reference/ExpsPlot.html">ExpsPlot</a></span><span class="op">(</span><span class="va">fit_Hotgenes</span>,</span>
<span>  xVar <span class="op">=</span> <span class="va">xvar</span>,</span>
<span>  yVar <span class="op">=</span> <span class="va">yvar</span>,</span>
<span>  filter_eval <span class="op">=</span> <span class="va">Hrs</span> <span class="op">!=</span> <span class="fl">2</span>,</span>
<span>  fill <span class="op">=</span> <span class="st">"Hrs"</span>,</span>
<span>  boxplot <span class="op">=</span> <span class="cn">TRUE</span></span>
<span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="Hotgenes_files/figure-html/exps_plot_2-1.png" width="576"></p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># Reorder data on the fly</span></span>
<span><span class="fu"><a href="../reference/ExpsPlot.html">ExpsPlot</a></span><span class="op">(</span><span class="va">fit_Hotgenes</span>,</span>
<span>  xVar <span class="op">=</span> <span class="va">xvar</span>,</span>
<span>  yVar <span class="op">=</span> <span class="va">yvar</span>,</span>
<span>  boxplot <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  fill <span class="op">=</span> <span class="st">"Hrs"</span>,</span>
<span>  named_levels  <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>Feature <span class="op">=</span> <span class="st">"IL6"</span>,</span>
<span>                       Hrs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"6"</span>, <span class="st">"2"</span>, <span class="st">"0"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="Hotgenes_files/figure-html/exps_plot_3-1.png" width="576"></p>
</div>
<div class="section level3">
<h3 id="visualize-high-magnitude-changes-hotgenes-with-a-heatmap">Visualize high magnitude changes (‘Hotgenes’) with a heatmap<a class="anchor" aria-label="anchor" href="#visualize-high-magnitude-changes-hotgenes-with-a-heatmap"></a>
</h3>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/DEphe.html">DEphe</a></span><span class="op">(</span><span class="va">fit_Hotgenes</span>,</span>
<span>      contrasts <span class="op">=</span> <span class="st">"sh_EWS_vs_Ctrl"</span>, </span>
<span>      Topn <span class="op">=</span> <span class="fl">5</span>,</span>
<span>      cellheight <span class="op">=</span> <span class="fl">10</span>,</span>
<span>      cellwidth <span class="op">=</span> <span class="fl">8</span>,</span>
<span>      annotation_colors <span class="op">=</span> <span class="fu"><a href="../reference/coldata_palettes.html">coldata_palettes</a></span><span class="op">(</span><span class="va">fit_Hotgenes</span><span class="op">)</span>,</span>
<span>      annotations <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Hrs"</span>, <span class="st">"sh"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="Hotgenes_files/figure-html/heatmap_1-1.png" width="768"></p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># change labels to ensembl_id</span></span>
<span><span class="fu"><a href="../reference/DEphe.html">DEphe</a></span><span class="op">(</span><span class="va">fit_Hotgenes</span>,</span>
<span>      contrasts <span class="op">=</span> <span class="st">"sh_EWS_vs_Ctrl"</span>,</span>
<span>      label_by <span class="op">=</span> <span class="st">"ensembl_id"</span>,</span>
<span>      Topn <span class="op">=</span> <span class="fl">5</span>,</span>
<span>      cellheight <span class="op">=</span> <span class="fl">10</span>,</span>
<span>      cellwidth <span class="op">=</span> <span class="fl">8</span>,</span>
<span>      annotation_colors <span class="op">=</span> <span class="fu"><a href="../reference/coldata_palettes.html">coldata_palettes</a></span><span class="op">(</span><span class="va">fit_Hotgenes</span><span class="op">)</span>,</span>
<span>      annotations <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Hrs"</span>, <span class="st">"sh"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="Hotgenes_files/figure-html/heatmap_2-1.png" width="768"></p>
</div>
<div class="section level3">
<h3 id="run-gsea-using-msigdbr-genesets">Run GSEA using msigdbr genesets<a class="anchor" aria-label="anchor" href="#run-gsea-using-msigdbr-genesets"></a>
</h3>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># get geneset</span></span>
<span><span class="va">H_paths</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GSEA_Support.html">msigdbr_wrapper</a></span><span class="op">(</span></span>
<span>    species <span class="op">=</span> <span class="st">"human"</span>,</span>
<span>    set <span class="op">=</span> <span class="st">"CP:KEGG"</span>,</span>
<span>    gene_col <span class="op">=</span> <span class="st">"gene_symbol"</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="co"># Get ranks</span></span>
<span>  <span class="va">InputRanks</span> <span class="op">&lt;-</span> <span class="va">fit_Hotgenes</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/utilities.html">DE</a></span><span class="op">(</span></span>
<span>      Report <span class="op">=</span> <span class="st">"Ranks"</span>,</span>
<span>      contrasts <span class="op">=</span> <span class="st">"Hrs_2_vs_0"</span>,</span>
<span>      Rank_name <span class="op">=</span> <span class="st">"Feature"</span>,</span>
<span>      padj_cut <span class="op">=</span> <span class="fl">0.1</span></span>
<span>    <span class="op">)</span></span>
<span>  </span>
<span>  </span>
<span>  <span class="co"># fgsea wrapper --------</span></span>
<span>  <span class="va">Out_GSEA</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GSEA_Support.html">fgsea_</a></span><span class="op">(</span></span>
<span>    Ranks <span class="op">=</span> <span class="va">InputRanks</span>,</span>
<span>    pathways <span class="op">=</span> <span class="va">H_paths</span>,</span>
<span>    nproc <span class="op">=</span> <span class="fl">1</span>,</span>
<span>    minSize <span class="op">=</span> <span class="fl">5</span>,</span>
<span>    maxSize <span class="op">=</span> <span class="cn">Inf</span></span>
<span>  <span class="op">)</span></span>
<span><span class="co">## [1] "Hrs_2_vs_0"</span></span>
<span><span class="co">##   |                                                                              |                                                                      |   0%  |                                                                              |===================================                                   |  50%  |                                                                              |======================================================================| 100%</span></span></code></pre></div>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>  </span>
<span>  <span class="co"># Get details for all</span></span>
<span>  <span class="va">Out_GSEA</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/GSEA_Support.html">fgsea_Results</a></span><span class="op">(</span></span>
<span>      contrasts <span class="op">=</span> <span class="st">"Hrs_2_vs_0"</span>,</span>
<span>      padj_cut <span class="op">=</span> <span class="fl">0.2</span>,</span>
<span>      mode <span class="op">=</span> <span class="st">"D"</span></span>
<span>    <span class="op">)</span></span>
<span><span class="co">## $Hrs_2_vs_0</span></span>
<span><span class="co">## <span style="color: #949494;"># A tibble: 5 × 9</span></span></span>
<span><span class="co">##   pathway             pval   padj log2err    ES   NES  size leadingEdge sign_NES</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;list&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> kegg_nod_like_r… 0.009<span style="text-decoration: underline;">69</span> 0.073<span style="text-decoration: underline;">7</span>   0.381 0.770  1.58     9 <span style="color: #949494;">&lt;chr [6]&gt;</span>          1</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> kegg_cytokine_c… 0.018<span style="text-decoration: underline;">4</span>  0.073<span style="text-decoration: underline;">7</span>   0.352 0.708  1.55    14 <span style="color: #949494;">&lt;chr [9]&gt;</span>          1</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> kegg_chemokine_… 0.051<span style="text-decoration: underline;">7</span>  0.135    0.219 0.734  1.44     7 <span style="color: #949494;">&lt;chr [7]&gt;</span>          1</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">4</span> kegg_jak_stat_s… 0.077<span style="text-decoration: underline;">9</span>  0.135    0.181 0.748  1.39     5 <span style="color: #949494;">&lt;chr [2]&gt;</span>          1</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">5</span> kegg_toll_like_… 0.084<span style="text-decoration: underline;">4</span>  0.135    0.174 0.742  1.38     5 <span style="color: #949494;">&lt;chr [3]&gt;</span>          1</span></span></code></pre></div>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>  </span>
<span>  <span class="co"># Or for one</span></span>
<span>  <span class="va">Out_GSEA</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/GSEA_Support.html">fgsea_Results</a></span><span class="op">(</span></span>
<span>      contrasts <span class="op">=</span> <span class="st">"Hrs_2_vs_0"</span>,</span>
<span>      padj_cut <span class="op">=</span> <span class="fl">0.2</span>,</span>
<span>      mode <span class="op">=</span> <span class="st">"leadingEdge"</span></span>
<span>    <span class="op">)</span></span>
<span><span class="co">## $Hrs_2_vs_0</span></span>
<span><span class="co">## $Hrs_2_vs_0$kegg_nod_like_receptor_signaling_pathway</span></span>
<span><span class="co">## [1] "IL6"     "CXCL8"   "TNFAIP3" "CXCL1"   "CCL2"    "CXCL2"  </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $Hrs_2_vs_0$kegg_cytokine_cytokine_receptor_interaction</span></span>
<span><span class="co">## [1] "IL6"    "CXCL8"  "CXCL1"  "IL11"   "CCL2"   "CXCL3"  "CXCL2"  "TGFBR1"</span></span>
<span><span class="co">## [9] "CXCR4" </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $Hrs_2_vs_0$kegg_chemokine_signaling_pathway</span></span>
<span><span class="co">## [1] "CXCL8" "CXCL1" "CCL2"  "CXCL3" "CXCL2" "CXCR4" "NFKB1"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $Hrs_2_vs_0$kegg_jak_stat_signaling_pathway</span></span>
<span><span class="co">## [1] "IL6"  "IL11"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $Hrs_2_vs_0$kegg_toll_like_receptor_signaling_pathway</span></span>
<span><span class="co">## [1] "IL6"   "CXCL8" "JUN"</span></span></code></pre></div>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>  </span>
<span>  <span class="co"># Generate a summary plot</span></span>
<span>  <span class="va">Out_GSEA</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/GSEA_Support.html">GSEA_Plots</a></span><span class="op">(</span></span>
<span>      contrasts <span class="op">=</span> <span class="st">"Hrs_2_vs_0"</span>,</span>
<span>      padj_cut <span class="op">=</span> <span class="fl">0.2</span>,</span>
<span>      width <span class="op">=</span> <span class="fl">30</span>,</span>
<span>      Topn <span class="op">=</span> <span class="fl">2</span></span>
<span>    <span class="op">)</span></span>
<span><span class="co">## $Hrs_2_vs_0</span></span></code></pre></div>
<p><img src="Hotgenes_files/figure-html/GSEA_1-1.png" width="576"></p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>  </span>
<span>  <span class="co"># plotEnrichment_</span></span>
<span>  <span class="fu"><a href="../reference/GSEA_Support.html">plotEnrichment_</a></span><span class="op">(</span></span>
<span>    <span class="va">Out_GSEA</span>, <span class="st">"Hrs_2_vs_0"</span>,</span>
<span>    <span class="st">"kegg_nod_like_receptor_signaling_pathway"</span></span>
<span>  <span class="op">)</span></span>
<span><span class="co">## leading edge genes for kegg_nod_like_receptor_signaling_pathway</span></span>
<span><span class="co">## c("IL6", "CXCL8", "TNFAIP3", "CXCL1", "CCL2", "CXCL2")</span></span></code></pre></div>
<p><img src="Hotgenes_files/figure-html/GSEA_1-2.png" width="576"></p>
</div>
<div class="section level3">
<h3 id="or-use-the-hotgenesets-function-for-gsva">Or use the HotgeneSets() function for gsva<a class="anchor" aria-label="anchor" href="#or-use-the-hotgenesets-function-for-gsva"></a>
</h3>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">choice_set</span> <span class="op">&lt;-</span> <span class="st">"CP:KEGG"</span></span>
<span><span class="va">choice_id</span> <span class="op">&lt;-</span> <span class="st">"gene_symbol"</span></span>
<span>  </span>
<span><span class="va">gsList</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GSEA_Support.html">msigdbr_wrapper</a></span><span class="op">(</span></span>
<span>    species <span class="op">=</span> <span class="st">"human"</span>,</span>
<span>    set <span class="op">=</span> <span class="va">choice_set</span>,</span>
<span>    gene_col <span class="op">=</span> <span class="va">choice_id</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="co"># HotgeneSets -------------------------------------------------------------</span></span>
<span>  </span>
<span>  <span class="va">HotgeneSets_out</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/HotgeneSets.html">HotgeneSets</a></span><span class="op">(</span></span>
<span>    Hotgenes <span class="op">=</span> <span class="va">fit_Hotgenes</span>,</span>
<span>    geneSets <span class="op">=</span> <span class="va">gsList</span>,</span>
<span>    kcdf <span class="op">=</span> <span class="st">"Gaussian"</span>,</span>
<span>    method <span class="op">=</span> <span class="st">"ssgsea"</span>,</span>
<span>    minSize <span class="op">=</span> <span class="fl">2</span>,</span>
<span>    maxSize <span class="op">=</span> <span class="cn">Inf</span></span>
<span>  <span class="op">)</span></span>
<span><span class="co">## [1] "using Feature col"</span></span>
<span><span class="co">## [1] "building mapper"</span></span>
<span><span class="co">## Estimating ssGSEA scores for 93 gene sets.</span></span>
<span><span class="co">## [1] "Calculating ranks..."</span></span>
<span><span class="co">## [1] "Calculating absolute values from ranks..."</span></span>
<span><span class="co">##   |                                                                              |                                                                      |   0%  |                                                                              |======                                                                |   8%  |                                                                              |============                                                          |  17%  |                                                                              |==================                                                    |  25%  |                                                                              |=======================                                               |  33%  |                                                                              |=============================                                         |  42%  |                                                                              |===================================                                   |  50%  |                                                                              |=========================================                             |  58%  |                                                                              |===============================================                       |  67%  |                                                                              |====================================================                  |  75%  |                                                                              |==========================================================            |  83%  |                                                                              |================================================================      |  92%  |                                                                              |======================================================================| 100%</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## [1] "Normalizing..."</span></span></code></pre></div>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>  </span>
<span>  </span>
<span>  <span class="va">HotgeneSets_out</span></span>
<span><span class="co">## class: Hotgenes </span></span>
<span><span class="co">## Original class/package:  EList/limma</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Differential expression (default thresholds): </span></span>
<span><span class="co">## |contrast       | total|</span></span>
<span><span class="co">## |:--------------|-----:|</span></span>
<span><span class="co">## |Hrs_2_vs_0     |    45|</span></span>
<span><span class="co">## |Hrs_6_vs_0     |    54|</span></span>
<span><span class="co">## |sh_EWS_vs_Ctrl |    29|</span></span>
<span><span class="co">## |shEWS.Hrs2     |     0|</span></span>
<span><span class="co">## |shEWS.Hrs6     |     0|</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Available feature mapping:  Feature, value </span></span>
<span><span class="co">## ExpressionSlots:  ssgsea </span></span>
<span><span class="co">## Total auxiliary assays:  2 </span></span>
<span><span class="co">## Total samples:  12</span></span></code></pre></div>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># store your Hotgenes objects in a named list</span></span>
<span><span class="co"># The Shiny_Hotgenes() will let you toggle between objects</span></span>
<span>  </span>
<span> </span>
<span><span class="kw">if</span><span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span><span class="op">{</span></span>
<span>  </span>
<span>  <span class="va">List_Hotgenes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>HotgeneSets_out <span class="op">=</span> <span class="va">HotgeneSets_out</span>,</span>
<span>                        fit_Hotgenes <span class="op">=</span> <span class="va">fit_Hotgenes</span><span class="op">)</span></span>
<span>  </span>
<span>   <span class="fu"><a href="../reference/Shiny_Hotgenes_UI.html">Shiny_Hotgenes</a></span><span class="op">(</span><span class="va">List_Hotgenes</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="all-functionality-is-available-in-a-shiny-app">All functionality is available in a shiny app!<a class="anchor" aria-label="anchor" href="#all-functionality-is-available-in-a-shiny-app"></a>
</h2>
<p>See manual for details!</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span><span class="op">{</span></span>
<span>  </span>
<span>  <span class="fu"><a href="../reference/Shiny_Hotgenes_UI.html">Shiny_Hotgenes</a></span><span class="op">(</span><span class="va">dds_Hotgenes</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Richard Virgen-Slane.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
